src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/MemberDefHandler.java

```diff
@@ -51,13 +53,38 @@ public void checkIndentation() {
             checkType();
         }
         final DetailAST firstNode = getMainAst();
-        final DetailAST lastNode = getVarDefStatementSemicolon(firstNode);
+        DetailAST lastNode = getVarDefStatementSemicolon(firstNode);
 
-        if (lastNode != null && !isArrayDeclaration(firstNode)) {
+        if (lastNode != null) {
+            lastNode = getArrayInitNode(firstNode).orElse(lastNode);
             checkWrappingIndentation(firstNode, lastNode);
         }
     }
 
+    /**
+     * Finds the array init node.
+     *
+     * @param firstNode Node to begin searching
+     * @return array init node
+     */
+    private static Optional<DetailAST> getArrayInitNode(DetailAST firstNode) {
+        Optional<DetailAST> result = Optional.empty();
+        final DetailAST assign = firstNode.findFirstToken(TokenTypes.ASSIGN);
+        if (assign != null) {
+            final DetailAST expr = assign.findFirstToken(TokenTypes.EXPR);
+            if (expr == null) {
+                result = Optional.ofNullable(assign.findFirstToken(TokenTypes.ARRAY_INIT));
```

Comment:
I am not very good with Optionals but there are alot of `null` checks in this method which is what optionals is to prevent.

This whole method cant be converted into 1 big optional that doesn't make things complex?